core/src/Kokkos_MemoryPool.hpp

```diff
@@ -606,7 +606,8 @@ class MemoryPool {
       //   0 <= sb_id
       //   sb_state_array == m_sb_state_array + m_sb_state_size * sb_id
 
-      if (sb_state == (state_header_mask & *sb_state_array)) {
+      // TODO memory order
```

Comment:
Is this TODONE?

Reply:
Sort of. The current semantics match those from before this change (`Impl::atomic_load` defaults to relaxed), but I'm not confident enough that `relaxed` is the right memory order here, so I put a TODO